crates/network-scanner/src/interfaces/windows.rs

```diff
@@ -0,0 +1,45 @@
+use std::net::IpAddr;
+
+use crate::interfaces::NetworkInterface;
+use anyhow::Context;
+
+pub fn get_network_interfaces() -> anyhow::Result<Vec<NetworkInterface>> {
+    ipconfig::get_adapters()
+        .context("Failed to get network interfaces")?
+        .into_iter()
+        .map(|adapter| Ok(adapter.into()))
+        .collect()
+}
+
+impl From<ipconfig::Adapter> for NetworkInterface {
+    fn from(adapter: ipconfig::Adapter) -> Self {
+        NetworkInterface {
+            name: adapter.adapter_name().to_string(),
+            description: Some(adapter.description().to_string()),
+            mac_address: adapter
+                .physical_address()
+                .map(|mac| vec![[mac[0], mac[1], mac[2], mac[3], mac[4], mac[5]]])
```

Comment:
issue: This is ignoring the existence of the 8-byte mac addresses. I believe that’s why the library is not returning a `[u8; 6]` directly.

Reply:
updated!